test/tests/performance/pbs_sched_perf.py

```diff
@@ -359,3 +361,105 @@ def test_attr_update_period_perf(self):
+                                {'scheduling': 'True'}, id=sched_name)
+
+    def setup_queues_nodes(self):
+        a = {'queue_type': 'execution',
+             'started': 'True',
+             'enabled': 'True'}
+        for i in range(1, 6):
+            self.server.manager(MGR_CMD_CREATE, QUEUE, a, id='wq' + str(i))
+            p = {'partition': 'P' + str(i)}
+            self.server.manager(MGR_CMD_SET, QUEUE, p, id='wq' + str(i))
+            node = str(self.mom.shortname)
+            num = i - 1
+            self.server.manager(MGR_CMD_SET, NODE, p,
+                                id=node + '[' + str(num) + ']')
+
+    def submit_njobs(self, num_jobs=1, attrs=None, user=TEST_USER):
+        """
+        Submit num_jobs number of jobs with attrs attributes for user.
+        Return a list of job ids
+        """
+        if attrs is None:
+            attrs = {ATTR_q: 'workq'}
+        ret_jids = []
+        for _ in range(num_jobs):
+            J = Job(user, attrs)
+            jid = self.server.submit(J)
+            ret_jids += [jid]
+
+        return ret_jids
+
+    @timeout(3600)
+    def test_multi_sched_perf(self):
+        """
+        Test time taken to schedule and run 5k jobs with
+        single scheduler and workload divided among 5 schedulers.
+        """
+        a = {'resources_available.ncpus': 1000}
+        self.server.create_vnodes(self.mom.shortname, a, 5, self.mom)
+        a = {'scheduling': 'False'}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+        self.submit_njobs(5000)
+        start = time.time()
+        self.scheduler.run_scheduling_cycle()
+        c = self.scheduler.cycles(lastN=1)[0]
+        cyc_dur = c.end - c.start
+        self.perf_test_result(cyc_dur, "default_cycle_duration", "secs")
+        msg = 'Time taken by default scheduler to run 5k jobs is '
+        self.logger.info(msg + str(cyc_dur))
+        self.server.cleanup_jobs()
+        self.setup_scheds()
+        self.setup_queues_nodes()
+        for sc in self.scheds:
+            a = {'scheduling': 'False'}
+            self.server.manager(MGR_CMD_SET, SCHED, a, id=sc)
+        a = {ATTR_q: 'wq1'}
+        self.submit_njobs(1000, a)
+        a = {ATTR_q: 'wq2'}
+        self.submit_njobs(1000, a)
+        a = {ATTR_q: 'wq3'}
+        self.submit_njobs(1000, a)
+        a = {ATTR_q: 'wq4'}
+        self.submit_njobs(1000, a)
+        a = {ATTR_q: 'wq5'}
+        self.submit_njobs(1000, a)
+        start = time.time()
+        for sc in self.scheds:
+            a = {'scheduling': 'True'}
+            self.server.manager(MGR_CMD_SET, SCHED, a, id=sc)
+        for sc in self.scheds:
+            a = {'scheduling': 'False'}
+            self.server.manager(MGR_CMD_SET, SCHED, a, id=sc)
+        for sc in self.scheds:
+            self.scheds[sc].log_match("Leaving Scheduling Cycle",
+                                      starttime=int(start),
+                                      max_attempts=30, interval=1)
```

Comment:
log_match() will return the matched line.  You can save a lot of log analyzing if you just call PBSLogUtils.convert_date_time(log.split(';')[0]) and take the largest value.  That's the end time of the longest cycle.

Reply:
@bhroam , I have to parse the starting scheduling cycle in that case to find the difference. Is it fine ?

Reply:
I don't see why you have to parse the scheduling cycle.  log_match() will return the 'Leaving Scheduling Cycle' line with the timestamp.  You pass the timestamp to convert_date_time() and it returns you a epoch time.  Pick the max of all the epoch times and you have your cycle end time.

Reply:
@bhroam Yes i can do that to findout cycle end time, but i want to calculate the cycle duration time , for that i need both start time and end time.
So i need both starting and leaving scheduling cycle time. so i want to log_match both the messages.

Reply:
You can find  the start by doing the log_match() or you'll likely be close enough if you just do a start_time = time.time() before you start kicking cycles.